synapse/util/retryutils.py

```diff
@@ -291,17 +294,15 @@ async def store_retry_timings() -> None:
                     self.retry_interval,
                 )
 
-                if self.notifier:
-                    # Inform the relevant places that the remote server is back up.
-                    self.notifier.notify_remote_server_up(self.destination)
-
-                if self.replication_client:
-                    # If we're on a worker we try and inform master about this. The
-                    # replication client doesn't hook into the notifier to avoid
-                    # infinite loops where we send a `REMOTE_SERVER_UP` command to
-                    # master, which then echoes it back to us which in turn pokes
-                    # the notifier.
-                    self.replication_client.send_remote_server_up(self.destination)
+                # If the server was previously failing, but is no longer.
+                if previously_failing:
```

Comment:
Actually, I think this is not quite right still, it will end up calling this code if we were previously failing & still failing. I think?

Reply:
> Actually, I think this is not quite right still, it will end up calling this code if we were previously failing & still failing. I think?

It should be OK now. 👍

Reply:
The logic *could* be simplified to only check `not currently_failing`, which depends on the earlier return to not kick off the background process. But I find this a bit clearer to include `previously_failing and not currently_failing`. 🤷